File: 0144/main.ts

function preorderTraversal(root: TreeNode | null): number[] {
    let array: number[] = []
    function order(node: TreeNode | null){
        if (node == null){
            return
        }
        array.push(node.val)
        order(node.left)
        order(node.right)
    }
    order(root)
    return array
};

File: 0144/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func preorderTraversal(root *TreeNode) []int {
	array := []int{}
	var order func(node *TreeNode)
	order = func(node *TreeNode) {
		if node == nil {
			return
		}
		array = append(array, node.Val)
		order(node.Left)
		order(node.Right)
	}
	order(root)
	return array
}
